src/components/Dialogs/Dialog/__tests__/Dialog.spec.tsx

```diff
@@ -0,0 +1,142 @@
+                contentLabel="Content Label"
+            >
+                {dialogBody}
+            </Dialog>
+        );
+    });
+
+    it('should render correctly with a title', () => {
+        expect(view.baseElement).toMatchSnapshot();
+        expect(screen.getByText(/Title/i)).toBeInTheDocument();
+        expect(screen.queryByRole('heading', { level: 2 })).toBeInTheDocument();
+        expect(screen.getByTestId('close-icon')).toBeInTheDocument();
+        expect(screen.getAllByRole('button')).toHaveLength(2);
+    });
+
+    it('should not render a title when one is not provided', () => {
+        view.rerender(
+            <Dialog
+                isOpen
+                acceptButton={acceptButtonProps}
+                cancelButton={cancelButtonProps}
+                onClose={mockOnCancel}
+                ariaHideApp={false}
+                contentLabel="Content Label"
+            >
+                {dialogBody}
+            </Dialog>
+        );
+        // screen.debug();
+        expect(view.baseElement).toMatchSnapshot();
+        expect(screen.queryByRole('heading', { level: 2 })).not.toBeInTheDocument();
+        expect(screen.getByTestId('close-icon')).toBeInTheDocument();
+        expect(screen.getAllByRole('button')).toHaveLength(2);
+    });
+
+    it('should not show buttons when no acceptButton has been passed', () => {
+        view.rerender(
+            <Dialog
+                isOpen
+                onClose={mockOnCancel}
+                ariaHideApp={false}
+                contentLabel="Content Label"
+                title={dialogTitle}
+            >
+                {dialogBody}
+            </Dialog>
+        );
+
+        expect(screen.getAllByRole('heading')).toHaveLength(1);
+        expect(screen.queryByRole('button')).not.toBeInTheDocument();
+    });
+
+    it('should render buttons with context critical state', () => {
+        view.rerender(
+            <Dialog
+                isOpen
+                onClose={mockOnCancel}
+                ariaHideApp={false}
+                contentLabel="Content Label"
+                variant="critical"
+                acceptButton={acceptButtonProps}
+                cancelButton={cancelButtonProps}
+            >
+                {dialogBody}
+            </Dialog>
+        );
+        expect(screen.getAllByRole('button')[0]).toHaveClass('Button--context_critical');
+    });
+
+    it('should render buttons with context primary state when no variant is passed', () => {
+        view.rerender(
+            <Dialog
+                isOpen
+                onClose={mockOnCancel}
+                ariaHideApp={false}
+                contentLabel="Content Label"
+                acceptButton={acceptButtonProps}
+                cancelButton={cancelButtonProps}
+            >
+                {dialogBody}
+            </Dialog>
+        );
+        expect(screen.getAllByRole('button')[0]).toHaveClass('Button--context_primary');
```

Comment:
can't we get the single button instead of querying by all and use the first one?